src/commonTest/kotlin/io/konform/validation/ValidationBuilderTest.kt

```diff
@@ -168,6 +170,95 @@ class ValidationBuilderTest {
         Register(email = "tester@").let { assertEquals(2, countFieldsWithErrors(splitDoubleValidation(it))) }
     }
 
+    @Test
+    fun functionAccessorSyntax() {
+        val splitDoubleValidation = Validation<Register> {
+            Register::getPasswordFun {
+                minLength(1)
+            }
+            Register::getPasswordFun {
+                maxLength(10)
+            }
+            Register::getEmailFun {
+                matches(".+@.+".toRegex())
+            }
+        }
+
+        Register(email = "tester@test.com", password = "a").let { assertEquals(Valid(it), splitDoubleValidation(it)) }
+        Register(email = "tester@test.com", password = "").let { assertEquals(1, countErrors(splitDoubleValidation(it), Register::password)) }
+        Register(email = "tester@test.com", password = "aaaaaaaaaaa").let { assertEquals(1, countErrors(splitDoubleValidation(it), Register::password)) }
+        Register(email = "tester@").let { assertEquals(2, countFieldsWithErrors(splitDoubleValidation(it))) }
+    }
+
+    @Test
+    fun lambdaAccessorSyntax() {
+        val splitDoubleValidation = Validation<Register> {
+            val getPassword = { r: Register -> r.password }
+            val getEmail = { r: Register -> r.email }
+            getPassword("getPasswordLambda") {
+                minLength(1)
+            }
```

Comment:
WDYT about this as an API for lambdas

```suggestion
            get("password", { it.password }) {
                minLength(1)
            }
```

I think that should be possible with
```kotlin
abstract fun <R> extract(name: String, prop: (T) -> R, init: ValidationBuilder<R>.() -> Unit): Constraint<T>
```